Observer/Product/AbstractChangedProductObserver.php

```diff
@@ -114,9 +114,18 @@ public function execute(Observer $observer)
      */
     protected function registerChangedItemStore(ProductInterface $product, int $storeId)
     {
-        $changedItem = $this->createChangedItem($product, $storeId);
-        if (!$this->changedItemRepository->exists($changedItem)) {
-            $this->changedItemRepository->save($changedItem);
+        $itemId = $product->getId();
+
+        $itemsToInsert = [$itemId];
+        $parentProduct = $this->configurableProductType->getParentIdsByChild($itemId);
+        if (count($parentProduct) > 0) {
+            $itemsToInsert = $parentProduct;
+        }
```

Comment:
Genial! Solamente un pequeño comentario, quizá es más claro llamar a la variable parentProducts ya que pueden ser varios?

```suggestion
        $parentProducts = $this->configurableProductType->getParentIdsByChild($itemId);
        if (count($parentProducts) > 0) {
            $itemsToInsert = $parentProducts;
        }
```